src/states/joining_peer.rs

```diff
@@ -234,11 +253,14 @@ impl Base for JoiningPeer {
     fn handle_direct_message(
         &mut self,
         msg: DirectMessage,
-        _pub_id: PublicId,
+        p2p_node: P2pNode,
         _outbox: &mut dyn EventBox,
     ) -> Result<Transition, RoutingError> {
-        debug!("{} Unhandled direct message: {:?}", self, msg);
-
+        debug!(
+            "{} Unhandled direct message, adding to backlog: {:?}",
+            self, msg
+        );
+        self.direct_msg_backlog.push((p2p_node, msg));
```

Comment:
Prevent duplicate `Relocate` message handling: either ignore `Relocate` here or add a check in `Elder::handle_relocate` to ignore the message if its `pub_id` is different from ours.

Reply:
Hm, given that relocation is not yet implicit, this is not an issue yet. I'll handle it myself.